parsers/nmparser/parse_lst_file.go

```diff
@@ -5,28 +5,31 @@ import (
 	"strings"
 )
 
-// LstData is the output struct from a lst file
-type LstData struct {
-	RunDetails              RunDetails
-	FinalParameterEstimates FinalParameterEstimates
-	FinalParameterStdErr    FinalParameterEstimates
-	ParameterStructures     ParameterStructures
-	ParameterNames          ParameterNames
-	OFV                     float64
-}
-
-func parseOFV(line string) float64 {
-	// get rid of everything but stars and ofv
-	result := line[10:]
-	result = strings.Replace(result, "*", "", -1)
-	output, _ := strconv.ParseFloat(strings.Fields(result)[0], 64)
-	return output
+func parseOFV(line string, ofvDetails *OfvDetails) {
+	if strings.Contains(line, "#OBJV") {
+		// get rid of everything but stars and ofv
+		result := line[10:]
+		result = strings.Replace(result, "*", "", -1)
+		output, _ := strconv.ParseFloat(strings.Fields(result)[0], 64)
+		ofvDetails.OFVNoConstant = output
+	} else if strings.Contains(line, "CONSTANT TO OBJECTIVE FUNCTION") {
+		ofvDetails.OFV, _ = strconv.ParseFloat(
+			strings.TrimSpace(strings.Replace(line, "N*LOG(2PI) CONSTANT TO OBJECTIVE FUNCTION:", "", -1)),
+			64)
+	} else if strings.Contains(line, "OBJECTIVE FUNCTION VALUE WITHOUT CONSTANT") {
+		ofvDetails.OFVNoConstant, _ = strconv.ParseFloat(
+			strings.TrimSpace(strings.Replace(line, "OBJECTIVE FUNCTION VALUE WITHOUT CONSTANT:", "", -1)),
+			64)
+	} else if strings.Contains(line, "OBJECTIVE FUNCTION VALUE WITH CONSTANT") {
+		ofvDetails.OFVWithConstant, _ = strconv.ParseFloat(
+			strings.TrimSpace(strings.Replace(line, "OBJECTIVE FUNCTION VALUE WITH CONSTANT:", "", -1)),
+			64)
+	}
 }
 
 // ParseLstEstimationFile parses the lst file
 func ParseLstEstimationFile(lines []string) LstData {
-
-	var ofvIndex int
+	ofvDetails := new(OfvDetails)
```

Comment:
why `new` it vs just declaring the type `var ofvDeatils OfvDetails`

Reply:
It's a pointer. The pattern is to call the "parseOFV" function repeatedly while looping through the lines, each time possibly updating one of the 3 values in the OfvDetails structure.